### PR TITLE
refactor(frontend): change folder structure

### PR DESCRIPTION
**Pull request by Bettelstab**
## 🍰 Pullrequest
Changes the folder structure in the following way:
- 'components' only contains shared components
- `pages/*` folders contain all the code only relevant to them
- components only used in `DefaultLayout` are colocated with that one in the `layouts` folder
- code from `utils` which contained project-specific constants and helper functions is now in folder `constants`
- `DreamMallButton` goes from `buttons` to `dream-mall-panel`
- `buttons` is renamed to `simple-button` (`MainButton` is unused and will be deleted by #2990)

Note: Folders in `(parenthesis)` are ignored by file system routing.

### CODE REVIEW

File: frontend/src/layouts/malltalk/invitation/IncomingInvitationItem.vue [hunk 1]
```diff
@@ -46,7 +46,7 @@
 </template>
 
 <script lang="ts" setup>
-import Invitation from '#components/malltalk/interfaces/Invitation'
+import Invitation from '#layouts/malltalk/interfaces/Invitation.js'
```
Comment (sebastian2357) [inline frontend/src/layouts/malltalk/invitation/IncomingInvitationItem.vue:49]: Why you changed it to ".js" ?

File: frontend/src/layouts/malltalk/invitation/TableSwap.vue [hunk 2]
```diff
@@ -13,8 +13,9 @@
 
 <script setup lang="ts">
 import InfoBox from '#components/info-box/InfoBox.vue'
-import Invitation from '#components/malltalk/interfaces/Invitation'
-import IncomingInvitationItem from '#components/malltalk/invitation/IncomingInvitationItem.vue'
+import Invitation from '#layouts/malltalk/interfaces/Invitation.js'
```
Comment (sebastian2357) [inline frontend/src/layouts/malltalk/invitation/TableSwap.vue:16]: Same here

File: frontend/src/layouts/malltalk/settings/TableSettings.vue [hunk 3]
```diff
@@ -19,7 +19,7 @@ import { useUserStore } from '#stores/userStore'
 import ChangeUsers from './ChangeUsers.vue'
 import TableSettingsRoot from './TableSettingsRoot.vue'
 
-import type MyTableSettings from '#components/malltalk/interfaces/MyTableSettings'
+import type MyTableSettings from '#layouts/malltalk/interfaces/MyTableSettings.js'
```
Comment (sebastian2357) [inline frontend/src/layouts/malltalk/settings/TableSettings.vue:22]: same here

File: frontend/src/layouts/malltalk/setup/TableSetup.vue [hunk 4]
```diff
@@ -23,7 +23,7 @@ import SelectUsers from './SelectUsers.vue'
 import StartSetup from './StartSetup.vue'
 import SubmitTable from './SubmitTable.vue'
 
-import type MyTableSettings from '#components/malltalk/interfaces/MyTableSettings'
+import type MyTableSettings from '#layouts/malltalk/interfaces/MyTableSettings.js'
```
Comment (sebastian2357) [inline frontend/src/layouts/malltalk/setup/TableSetup.vue:26]: same here

**Comment by Bettelstab** [comment]
Ooops. That's what my VS Code does when it auto-imports, don't know how I can change it.

**Pull request merged**